pkg/gameserverallocations/metrics.go

```diff
@@ -123,3 +142,9 @@ func (r *metrics) setResponse(o k8sruntime.Object) {
 func (r *metrics) record() {
 	stats.Record(r.ctx, gameServerAllocationsLatency.M(time.Since(r.start).Seconds()))
 }
+
+// record the current allocation retry rate.
+func (r *metrics) recordAllocationErrorRate(ctx context.Context, retryCount int) {
```

Comment:
If I understand correctly the metrics is tracking the retry count when the retry succeed. So the name of the function is a little confusing since it is not error rate.

Reply:
I can change it to `recordAllocationRetryRate`